### PR TITLE
Fix cmul

### PR DESCRIPTION
**Pull request by lopelopelope**
fixed https://github.com/intel-analytics/BigDL/issues/1697

### CODE REVIEW

File: spark/dl/src/test/scala/com/intel/analytics/bigdl/nn/CMulSpec.scala [hunk 1]
```diff
@@ -44,6 +43,11 @@ class CMulSpec extends FlatSpec with Matchers {
     val gradInput1 = layer1.backward(input, gradOutput)
     val output2 = layer2.forward(input)
     val gradInput2 = layer2.backward(input, gradOutput)
+    val cmul = CMul[Float](Array[Int](1, 4096, 1, 1))
```
Comment (yiheng) [inline spark/dl/src/test/scala/com/intel/analytics/bigdl/nn/CMulSpec.scala]: This is not regular test code. You should verify the result.

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/CMul.scala [hunk 2]
```diff
@@ -82,7 +82,10 @@ class CMul[T: ClassTag](
   }
 
   private def mulOneDimWeight(dim: Int, expand: Tensor[T], output: Tensor[T]): Unit = {
-    val (innerNum, outerNum) = Utils.getInnerOuterNum(dim, output)
+    val multiplyDimSize = expand.size(dim)
+    var dimOfoutput : Int = 1
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/CMul.scala]: dimOfoutput  -> dimOfOutput

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/CMul.scala [hunk 3]
```diff
@@ -82,7 +82,10 @@ class CMul[T: ClassTag](
...
+    while(output.size(dimOfoutput) != multiplyDimSize) dimOfoutput += 1
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/CMul.scala]: What if there's no multiplyDimSize in output.size()? TTry to catch the array outofbound error and throw a more meaningful error msg

**Comment by yiheng** [comment]
What about the jenkins test result?

**Comment by lopelopelope** [comment]
It is still under test. I will contact you when it is done @yiheng

**Comment by lopelopelope** [comment]
pass jenkins @yiheng

**Pull request merged**